tests/tests/test-xcm/test-xcm-erc20-fees-and-trap.ts

```diff
@@ -0,0 +1,348 @@
+      })
+    );
+    expectEVMResult(result2.events, "Succeed");
+
+    // Check the sovereign account has reveived erc20 tokens
+    expect(
+      (
+        await web3EthCall(context.web3, {
+          to: erc20ContractAddress,
+          data: ERC20_INTERFACE.encodeFunctionData("balanceOf", [paraSovereign]),
+        })
+      ).result
+    ).equals(bnToHex(amountTransferred, { bitLength: 256 }));
+
+    // Create xcm message to send ERC20 tokens to Charleth
+    const config = {
+      assets: [
+        {
+          multilocation: {
+            parents: 0,
+            interior: {
+              X1: { PalletInstance: balancesPalletIndex },
+            },
+          },
+          fungible: 1_000_000_000_000_000n,
+        },
+        {
+          multilocation: {
+            parents: 0,
+            interior: {
+              X2: [
+                {
+                  PalletInstance: erc20XcmPalletIndex,
+                },
+                {
+                  AccountKey20: {
+                    network: "Any",
+                    key: erc20ContractAddress,
+                  },
+                },
+              ],
+            },
+          },
+          fungible: amountTransferred,
+        },
+      ],
+      beneficiary: CHARLETH_ADDRESS,
+    };
+
+    // Build the xcm message without deposit_asset()
+    // This is to trap all the assets present in the holding register
+    const xcmMessage = new XcmFragment(config)
+      .withdraw_asset()
+      .clear_origin()
+      .buy_execution()
+      .as_v2();
+
+    // Mock the reception of the xcm message
+    await injectHrmpMessage(context, paraId, {
+      type: "XcmVersionedXcm",
+      payload: xcmMessage,
+    } as RawXcmMessage);
+    await context.createBlock();
+
+    const amountOfTrappedAssets = 876_639_200_000_000n;
+    const claimConfig = {
+      assets: [
+        {
+          multilocation: {
+            parents: 0,
+            interior: {
+              X1: { PalletInstance: balancesPalletIndex },
+            },
+          },
+          fungible: amountOfTrappedAssets,
+        },
+      ],
+      beneficiary: paraSovereign,
+    };
+    // Check non-erc20 can be claimed
+    const xcmMessageToClaimAssets = new XcmFragment(claimConfig)
+      .claim_asset()
+      .buy_execution()
+      .deposit_asset()
+      .as_v2();
+
+    // Mock the reception of the xcm message
+    await injectHrmpMessage(context, paraId, {
+      type: "XcmVersionedXcm",
+      payload: xcmMessageToClaimAssets,
+    } as RawXcmMessage);
+    await context.createBlock();
+
+    // Search for AssetsClaimed event
+    const records = (await context.polkadotApi.query.system.events()) as any;
+    const events = records.filter(
+      ({ event }) => event.section == "polkadotXcm" && event.method == "AssetsClaimed"
+    );
+    expect(events).to.have.lengthOf(1);
+
+    const feePaidForClaimingAssets = 30_962_750_000_000n;
```

Comment:
Please compute the fee instead of using an ahardcoded value, you can do that in two steps: 

1. get the message weights with `weightMessage`
2. Mullitply the weight by a constant, the formula is: `const chargedFee = chargedWeight * 50000n;`